src/frontend/src/lib/derived/user-profile.derived.ts

```diff
@@ -1,14 +1,18 @@
 import type { Settings } from '$declarations/backend/backend.did';
 import { userProfileStore } from '$lib/stores/user-profile.store';
 import type { Option } from '$lib/types/utils';
-import { fromNullable, nonNullish } from '@dfinity/utils';
+import { fromNullable, isNullish, nonNullish } from '@dfinity/utils';
 import { derived, type Readable } from 'svelte/store';
 
 export const userSettings: Readable<Option<Settings>> = derived(
 	[userProfileStore],
 	([$userProfile]) =>
-		// The user profile may or may not have the settings field.
-		// To distinguish between user profile not loaded and missing settings,
-		// we use `undefined` for the former and `null` for the latter.
-		nonNullish($userProfile) ? (fromNullable($userProfile.profile.settings) ?? null) : undefined
+		nonNullish($userProfile) ? fromNullable($userProfile.profile.settings) : undefined
+);
+
+export const userSettingsLoaded: Readable<boolean> = derived(
+	[userProfileStore, userSettings],
+	([$userProfile, $userSettings]) =>
+		nonNullish($userSettings) ||
```

Comment:
Would `fromNullable($userProfile?.profile.settings) !== undefined` do the tricks?

Reply:
the only case where it does not work that is when `userProfile` is loaded but `settings` are missing (old accounts). In that case it would be `fromNullable([]) !== undefined` that returns `false`, but we need `true`: it is ok that the `settings` are missing when the `userProfile` is loaded.

Reply:
Gotcha. Then we can follow your approach but, can we resolve it without a dependency on `$userSettings`? I feel like we spare one evaluation if we solely use `$userProfileStore`

Reply:
in that case it will just be a `userProfileLoaded` check, is that ok? if the user profile is loaded, we don't care if settings are nullish or not, they are loaded

I had put that dependency just to not have that small amount of time where `userProfile` is not nullish and `userSettings` is still calculating; hopefully it does not impact that much, right?

Reply:
Ah yeah actually that should do, there is no use case where the user profile is loaded but the settings arent, good idea 👍